src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -199,8 +199,19 @@ private Binder GetEnclosingBinder(CSharpSyntaxNode node, int position)
                 }
                 else if (current.Kind() == SyntaxKind.TypeOfExpression && typeOfArgument == null)
                 {
-                    typeOfArgument = ((TypeOfExpressionSyntax)current).Type;
-                    typeOfEncounteredBeforeUnexpectedAnonymousFunction = unexpectedAnonymousFunction == null;
+                    var currentTypeofExpr = (TypeOfExpressionSyntax)current;
+
+                    // A span that contains everything inbetween the parenthesis in a typeof expression.
+                    // typeof   (    ThisIsAType       )
+                    //           |--------------------|
+                    var insideParensSpan = TextSpan.FromBounds(
+                        currentTypeofExpr.OpenParenToken.Span.Start + 1,
+                        currentTypeofExpr.CloseParenToken.Span.End);
+
+                    if (insideParensSpan.Contains(position)) {
```

Comment:
Reviewers: This check should in theory be a part of the `else if` directly above.  However, I avoided it for style reasons.  This means that the Debug.Assert(..) call in the `else` case doesn't execute.  Should this code:

1. Be moved entirely into the `else if`
2. Put the span up above the whole `if` chain
3. Keep as-is but with an extra Debug.Assert
4. Just keep as-is